composer/algorithms/cutout/README.md

```diff
@@ -0,0 +1,100 @@
+# ✂️ Cutout
+
+[\[How to Use\]](#how-to-use) - [\[Suggested Hyperparameters\]](#suggested-hyperparameters) - [\[Technical Details\]](#technical-details) - [\[Attribution\]](#attribution)
+
+`Computer Vision`
+
+Cutout is a data augmentation technique that masks out one or more square regions of an input image, replacing them with gray boxes.
+It is a regularization technique that improves the accuracy of models for computer vision.
+
+| ![CutOut](cut_out.png) |
+|:--:
+|*Several images from the CIFAR-10 dataset with Cutout applied. Cutout adds a gray box that occludes a portion of each image. This is [Figure 1 from DeVries & Taylor (2017)](https://arxiv.org/abs/1708.04552).*|
+
+## How to Use
+
+### Functional Interface
+
+TODO(CORY): FIX
+
+```python
+def training_loop(model, train_loader):
+  opt = torch.optim.Adam(model.parameters())
+  loss_fn = F.cross_entropy
+  model.train()
+  
+  for epoch in range(num_epochs):
+      for X, y in train_loader:
+          y_hat = model(X)
+          loss = loss_fn(y_hat, y)
+          loss.backward()
+          opt.step()
+          opt.zero_grad()
+```
+
+### Composer Trainer
+
+TODO(CORY): Fix and provide commentary and/or comments
+
+```python
+from composer.algorithms import XXX
+from composer.trainer import Trainer
+
+trainer = Trainer(model=model,
+                  train_dataloader=train_dataloader,
+                  max_duration='1ep',
+                  algorithms=[
+                  ])
+
+trainer.fit()
+```
+
+### Implementation Details
+
+TODO(CORY): Briefly describe how this is implemented under the hood in Composer. Need to explain the mask in particular, since it shows up a few times below.
```

Comment:
CutMix randomly selects `n_holes` square regions (which are possibly overlapping) with side length `length` and uses them to generate a binary mask for the image where a point within any hole is set to 0 and the remaining points are set to 1. This mask is then multiplied element-wise with the image in order to set the pixel value of any pixel value within a hole to 0.